.github/workflows/ci.yml

```diff
@@ -0,0 +1,300 @@
+    # Ubuntu has out of date tools and sometimes system updates
+    # require manual interaction so its unusable for CI
+    container: archlinux/base
+
+    steps:
+      # 1. Update the system
+      #    Arch linux updates NEVER require manual interaction,
+      #    so this is safe in combination with --noconfirm
+      - name: Update System
+        run: pacman --noconfirm -Syu
+      # 2. Install the required dependencies
+      - name: Install dependencies
+        run: pacman --noconfirm -S cmake make binutils gcc python3 git vulkan-devel ninja tar unzip which
+      # 3. Checkout the repository, this is done via a github action
+      - name: Checkout repository
+        uses: actions/checkout@master
+      # 4. Prepare nova
+      #    Currently there are 2 different build methods in use with nova,
+      #    so check which one the branch we are on is using and adjust as required
+      - name: Prepare
+        shell: bash
+        run: |
+          set -e
+
+          if [[ -f "build.py" ]]; then
+              # The new build.py method is easy, just run it with python
+              # and we are good to go
+              python3 build.py
+          else
+              # The fiddly part, this will require a manual install of vcpkg... lets do it
+              git clone https://github.com/microsoft/vcpkg vcpkg
+              pushd vcpkg
+              bash bootstrap-vcpkg.sh
+              export PATH=`pwd`:$PATH
+              popd
+              # Now vcpkg should be set up
+              VULKAN_SDK=/usr bash setup-linux.sh
+          fi
+      # 5. Configure with CMake
+      - name: Configure with CMake
+        run: |
+          mkdir build
+          cd build
+          VULKAN_SDK=/usr cmake ..   \
+            -G"Unix Makefiles"       \
+            -DNOVA_TEST=ON           \
+            -DCMAKE_BUILD_TYPE=Debug \
+            -DCMAKE_TOOLCHAIN_FILE=../vcpkg/scripts/buildsystems/vcpkg.cmake
+      # 6. Build with CMake
+      #    We use 3 threads on 2 cores since thats what Github provides as of writing this script,
+      #    for further reference see https://help.github.com/en/articles/virtual-environments-for-github-actions
+      - name: Build with CMake
+        working-directory: build
+        run: cmake --build . -- -j3
+      # 7. Run the unit tests
+      - name: Run unit tests
+        working-directory: build
+        run: ./nova-test-unit --gtest_color=yes
+      # 8. Prepare artifacts for uploading
+      - name: Prepare artifacts
+        run: |
+          # Binary artifacts
+          mkdir artifacts
+          cp LICENSE artifacts
+          cp build/nova-test-unit artifacts
+          cp build/nova-test-end-to-end artifacts
+          cp build/libnova-renderer.a artifacts
+      # 9. Upload binary artifacts
+      - name: Upload binary artifacts
+        uses: actions/upload-artifact@master
+        with:
+          name: binary-linux-gcc
+          path: artifacts
+
+  build-linux-clang:
```

Comment:
This duplication can be solved with a build matrix. See https://github.com/BVE-Reborn/bve-reborn/blob/master/.github/workflows/build.yml for how I did it across OSs, you can use the same technique for across compilers.